src/main/webapp/js/index.js

```diff
@@ -12,6 +12,79 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** Fetch receipts from the server and add them to the DOM. */
+async function searchReceipts() {
+  const label = document.getElementById('search-input').value;
+  const response = await fetch(`/search-receipts?label=${label}`);
+  const receipts = await response.json();
+
+  displayReceipts(label, receipts);
+}
+
+/* Clear out old receipt display and populate with new queried receipts. */
+function displayReceipts(label, receipts) {
+  const receiptsDisplayed = document.getElementById('receipt-list');
+  receiptsDisplayed.innerHTML = '';
+
+  // If no receipts returned, display an error message. Else, display receipts.
+  if (Object.keys(receipts).length === 0) {
+    receiptsDisplayed.innerHTML =
+        '<div class="col-md-12 text-center error-message"><h3>' +
+        'Sorry, no results found for "' + label +
+        '". Please try your search again or try a different query.</h3></div>';
+  } else {
+    receipts.forEach((receipt) => {
+      receiptsDisplayed.innerHTML += createReceiptCardElement(receipt);
+    });
+  }
+}
+
+
+/**
+ * Returns an HTML string that represents a receipt card.
+ * This card displays transaction date, store name, trasaction total,
+ * categories, receipt photo, and view/edit/delete buttons.
+ */
+function createReceiptCardElement(receipt) {
+  const categories = Array.from(receipt.categories);
+  const formattedHTML = '<div class="col-md-6">' +
```

Comment:
I might try to put this in the html file itself using the `template` tag (See https://www.w3schools.com/tags/tag_template.asp) and use `document.getElementById()` to fill in the template.

From https://google.github.io/styleguide/htmlcssguide.html#Separation_of_Concerns :

Separate structure from presentation from behavior.
Strictly keep structure (markup), presentation (styling), and behavior (scripting) apart, and try to keep the interaction between the three to an absolute minimum.

Reply:
I was looking for something like this but I guess I wasn't googling the right thing! Thanks